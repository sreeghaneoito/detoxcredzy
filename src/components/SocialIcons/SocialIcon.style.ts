import {StyleSheet} from 'react-native';

export const style = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 20,
    justifyContent: 'center',
    position: 'relative',
    paddingBottom: 40,
  },
  btn: {
    backgroundColor: '#28AF51',
    width: 25,
    height: 25,
    borderRadius: 20,
    marginHorizontal: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  back: {
    position: 'absolute',
    left: 20,
    top: 20,
  },
});
