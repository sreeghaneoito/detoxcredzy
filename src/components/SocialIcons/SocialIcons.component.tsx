import {View, Pressable} from 'react-native';
import React from 'react';
// import Instagram from '../../assets/icons/Instagram';
// import Facebook from '../../assets/icons/Facebook';
// import Twitter from '../../assets/icons/Twitter';
import {style} from './SocialIcon.style';
import LeftArrow from '../../assets/icons/LeftArrow';
// import {useNavigation} from '@react-navigation/native';
import {useUtils} from '../../hooks/useUtils';
import {routeEnum} from '../../enums/route.enum';
const SocialIcons = () => {
  const {gotoScreen} = useUtils();

  // * this will be in use after the client give the details
  // const socialURL = {
  //   instagram: 'https://instagram.com/',
  //   facebook: 'https://facebook.com/',
  //   twitter: 'https://twitter.com/',
  // };
  // const openURL = async (url: string) => {
  //   await Linking.openURL(url);
  // };
  return (
    <View style={style.container}>
      <Pressable
        style={style.back}
        onPress={() => {
          gotoScreen(routeEnum.DASHBOARDTAB);
        }}
        testID="goback">
        <LeftArrow />
      </Pressable>
      {/* <Pressable
        style={style.btn}
        testID="instagram"
        onPress={() => {
          openURL(socialURL.instagram);
        }}>
        <Instagram />
      </Pressable>
      <Pressable
        style={style.btn}
        testID="facebook"
        onPress={() => {
          openURL(socialURL.facebook);
        }}>
        <Facebook />
      </Pressable>
      <Pressable
        style={style.btn}
        testID="twitter"
        onPress={() => {
          openURL(socialURL.twitter);
        }}>
        <Twitter />
      </Pressable> */}
    </View>
  );
};

export default SocialIcons;
