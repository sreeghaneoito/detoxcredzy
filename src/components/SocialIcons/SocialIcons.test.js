import React from 'react';
import renderer from 'react-test-renderer';
import SocialIcons from './SocialIcons.component';

test('renders correctly', () => {
  const socialicons = renderer.create(<SocialIcons />).toJSON();
  expect(socialicons).toMatchSnapshot();
});
