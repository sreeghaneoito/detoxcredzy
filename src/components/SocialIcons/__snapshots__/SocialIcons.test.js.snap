// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "padding": 20,
      "position": "relative",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "left": 20,
        "position": "absolute",
        "top": 20,
      }
    }
    testID="goback"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="24px"
      bbWidth="24px"
      fill="none"
      focusable={false}
      height="24px"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 24,
            "width": 24,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="24px"
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M14.5303 18.5303C14.2374 18.8232 13.7626 18.8232 13.4697 18.5303L7.46967 12.5303C7.17678 12.2374 7.17678 11.7626 7.46967 11.4697L13.4697 5.46967C13.7626 5.17678 14.2374 5.17678 14.5303 5.46967C14.8232 5.76256 14.8232 6.23744 14.5303 6.53033L9.06066 12L14.5303 17.4697C14.8232 17.7626 14.8232 18.2374 14.5303 18.5303Z"
          fill={4280164130}
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#28AF51",
        "borderRadius": 20,
        "height": 25,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "width": 25,
      }
    }
    testID="instagram"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16px"
      bbWidth="16px"
      focusable={false}
      height="16px"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="16px"
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M11.9999802,2 C9.28416306,2 8.94357942,2.01151157 7.87701279,2.06017771 C6.81258969,2.10872478 6.0856937,2.27778605 5.44958032,2.5250069 C4.79203161,2.78052401 4.23431598,3.12245728 3.67838663,3.67838663 C3.12245728,4.23435568 2.78052401,4.79203161 2.5249672,5.44962002 C2.27774635,6.0856937 2.10872478,6.81262938 2.06013802,7.87701279 C2.01147187,8.94361912 2,9.28412336 2,11.9999802 C2,14.7158369 2.01147187,15.0563809 2.06013802,16.1229872 C2.10872478,17.1873706 2.27774635,17.9143063 2.5249672,18.55038 C2.78052401,19.2079287 3.12245728,19.7656443 3.67838663,20.3216134 C4.23431598,20.8775427 4.79203161,21.2194363 5.44958032,21.4750328 C6.0856937,21.722214 6.81258969,21.8912752 7.87701279,21.939862 C8.94357942,21.9884884 9.28416306,22 11.9999802,22 C14.7158369,22 15.0563412,21.9884884 16.1229475,21.939862 C17.1873706,21.8912752 17.9142666,21.722214 18.55038,21.4750328 C19.2079684,21.2194363 19.7656443,20.8775427 20.3215737,20.3216134 C20.8775427,19.7656443 21.2194363,19.2079684 21.4749931,18.55038 C21.722214,17.9143063 21.8912752,17.1873706 21.9397826,16.1229872 C21.9884884,15.0563809 22,14.7158369 22,11.9999802 C22,9.28416306 21.9884884,8.94361912 21.9397826,7.87701279 C21.8912752,6.81262938 21.722214,6.0856937 21.4749931,5.44962002 C21.2194363,4.79203161 20.8775427,4.23435568 20.3215737,3.67838663 C19.7656443,3.12245728 19.2079684,2.78052401 18.55038,2.5250069 C17.9142666,2.27778605 17.1873706,2.10872478 16.1229475,2.06017771 C15.0563412,2.01151157 14.7158369,2 11.9999802,2 M11.9999802,3.80179858 C14.6701082,3.80179858 14.9863588,3.81196052 16.0408184,3.86007094 C17.0158086,3.9045691 17.545301,4.06747764 17.8976741,4.2044256 C18.3644483,4.38583204 18.697609,4.60252738 19.0475209,4.95243936 C19.3974329,5.30239103 19.6141283,5.635512 19.7955744,6.10228624 C19.9324827,6.45465931 20.0953912,6.98419144 20.1398894,7.95914187 C20.1879601,9.01360151 20.1981617,9.32989177 20.1981617,11.9999802 C20.1981617,14.6701082 20.1879601,14.9863985 20.1398894,16.0408184 C20.0953912,17.0158483 19.9324827,17.5453407 19.7955744,17.8977138 C19.6141283,18.3644483 19.3974329,18.697609 19.0475209,19.0475209 C18.697609,19.3974726 18.3644483,19.614168 17.8976741,19.7955744 C17.545301,19.9324827 17.0158086,20.0954309 16.0408184,20.1398894 C14.9865176,20.1879998 14.6703067,20.1982014 11.9999802,20.1982014 C9.3296536,20.1982014 9.01344273,20.1879998 7.95914187,20.1398894 C6.98415175,20.0954309 6.45461961,19.9324827 6.10224654,19.7955744 C5.635512,19.614168 5.30235134,19.3974726 4.95247906,19.0475209 C4.60252738,18.697609 4.38579234,18.3644483 4.20438591,17.8977138 C4.06747764,17.5453407 3.9045691,17.0158483 3.86007094,16.0408581 C3.81196052,14.9863985 3.80179858,14.6701082 3.80179858,11.9999802 C3.80179858,9.32989177 3.81196052,9.01360151 3.86007094,7.95914187 C3.9045691,6.98419144 4.06747764,6.45465931 4.20438591,6.10228624 C4.38579234,5.635512 4.60252738,5.30239103 4.95243936,4.95247906 C5.30235134,4.60252738 5.635512,4.38583204 6.10224654,4.2044256 C6.45461961,4.06747764 6.98415175,3.9045691 7.95910218,3.86007094 C9.01360151,3.81196052 9.32985208,3.80179858 11.9999802,3.80179858 M11.9999802,15.3333333 C10.1590422,15.3333333 8.66662697,13.8409181 8.66662697,11.9999802 C8.66662697,10.1590422 10.1590422,8.66666667 11.9999802,8.66666667 C13.8409181,8.66666667 15.3333333,10.1590422 15.3333333,11.9999802 C15.3333333,13.8409181 13.8409181,15.3333333 11.9999802,15.3333333 M11.9999802,6.86486808 C9.16392671,6.86486808 6.86482839,9.16392671 6.86482839,11.9999802 C6.86482839,14.8360733 9.16392671,17.1351319 11.9999802,17.1351319 C14.8360336,17.1351319 17.1350922,14.8360733 17.1350922,11.9999802 C17.1350922,9.16392671 14.8360336,6.86486808 11.9999802,6.86486808 M18.5379991,6.66197868 C18.5379991,7.32472744 18.0007661,7.86196042 17.3380174,7.86196042 C16.6752686,7.86196042 16.1379959,7.32472744 16.1379959,6.66197868 C16.1379959,5.99922992 16.6752686,5.46195724 17.3380174,5.46195724 C18.0007661,5.46195724 18.5379991,5.99922992 18.5379991,6.66197868"
          fill={4294967295}
          name="Shape"
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#28AF51",
        "borderRadius": 20,
        "height": 25,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "width": 25,
      }
    }
    testID="facebook"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16px"
      bbWidth="16px"
      focusable={false}
      height="16px"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="16px"
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M13.397 20.997v-8.196h2.765l.411-3.209h-3.176V7.548c0-.926.258-1.56 1.587-1.56h1.684V3.127A22.336 22.336 0 0 0 14.201 3c-2.444 0-4.122 1.492-4.122 4.231v2.355H7.332v3.209h2.753v8.202h3.312z"
          fill={4294967295}
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#28AF51",
        "borderRadius": 20,
        "height": 25,
        "justifyContent": "center",
        "marginHorizontal": 5,
        "width": 25,
      }
    }
    testID="twitter"
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight="16px"
      bbWidth="16px"
      focusable={false}
      height="16px"
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={24}
      vbWidth={24}
      width="16px"
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M8.28251301,20.262505 C15.8295318,20.262505 19.9591837,14.0040016 19.9591837,8.58583433 C19.9591837,8.40976391 19.9591837,8.23369348 19.9511805,8.05762305 C20.7515006,7.48139256 21.4477791,6.75310124 22,5.92877151 C21.2637055,6.25690276 20.4713886,6.4729892 19.6390556,6.57703081 C20.487395,6.07282913 21.1356543,5.2645058 21.4477791,4.30412165 C20.6554622,4.77631052 19.77511,5.11244498 18.8387355,5.29651861 C18.0864346,4.49619848 17.0220088,4 15.8455382,4 C13.5806323,4 11.739896,5.84073629 11.739896,8.10564226 C11.739896,8.42577031 11.779912,8.73789516 11.8439376,9.04201681 C8.43457383,8.87394958 5.40936375,7.23329332 3.38455382,4.75230092 C3.03241297,5.36054422 2.83233293,6.06482593 2.83233293,6.81712685 C2.83233293,8.24169668 3.56062425,9.49819928 4.65706283,10.2344938 C3.98479392,10.2104842 3.35254102,10.0264106 2.80032013,9.72228892 L2.80032013,9.77831132 C2.80032013,11.7631052 4.21688675,13.4277711 6.08963585,13.8039216 C5.7454982,13.89996 5.38535414,13.9479792 5.00920368,13.9479792 C4.74509804,13.9479792 4.4889956,13.9239696 4.24089636,13.8759504 C4.76110444,15.5086034 6.28171269,16.6930772 8.07442977,16.72509 C6.66586635,17.8295318 4.89715886,18.4857943 2.97639056,18.4857943 C2.6482593,18.4857943 2.32012805,18.4697879 2,18.4297719 C3.80872349,19.5822329 5.96958784,20.262505 8.28251301,20.262505"
          fill={4294967295}
          name="Shape"
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
